Backend/WebApi-PointMap/Controllers/UserController.cs

```diff
@@ -2,69 +2,41 @@
 using ManagerLayer.Login;
 using DTO;
 using System;
-using System.Collections.Generic;
-using System.Data.Entity.Validation;
-using System.Linq;
 using System.Net;
-using System.Net.Http;
 using System.Web.Http;
-using System.Web.Http.Cors;
 using WebApi_PointMap.Models;
-using System.Web.Http.Controllers;
-using static ServiceLayer.Services.ExceptionService;
 using ManagerLayer.UserManagement;
+using WebApi_PointMap.ErrorHandling;
 
 namespace WebApi_PointMap.Controllers
 {
     public class UserController : ApiController
     {
         UserLoginManager _userLoginManager;
-        UserManagementManager _userManagementManager;
+        DatabaseContext _db;
 
         // POST api/user/login
         [HttpPost]
         [Route("api/user/login")]
         public IHttpActionResult LoginFromSSO([FromBody] LoginDTO requestPayload)
         {
-            if (!ModelState.IsValid || requestPayload == null)
-            {
-                return Content((HttpStatusCode)412, ModelState);
-            }
-            _userLoginManager = new UserLoginManager();
-            Guid userSSOID;
             try
             {
-                // check if valid SSO ID format
-                userSSOID = Guid.Parse(requestPayload.SSOUserId);
-            }
-            catch (Exception)
-            {
-                return Content((HttpStatusCode)400, "Invalid SSO ID");
-            }
-            using (var _db = new DatabaseContext())
-            {
+                //throws ExceptionService.InvalidModelPayloadException
+                ControllerHelpers.ValidateModelAndPayload(ModelState, requestPayload);
+                
+                //throws ExceptionService.InvalidGuidException
+                var userSSOID = ControllerHelpers.ParseAndCheckId(requestPayload.SSOUserId);
+                _db = new DatabaseContext();

```

Comment:
create constructor for class, initalize _db in constructor, make _db global variable as done in other controller. Same with the managers used

Reply:
Completed